app/Http/Controllers/TeamsController.php

```diff
@@ -66,6 +74,28 @@ public function edit(string $id): Response
         return ext_view('teams.edit', compact('team'));
     }
 
+    public function leaderboard(string $id, ?string $ruleset = null): Response
+    {
+        $team = Team::findOrFail($id);
+        $ruleset ??= Beatmap::modeStr($team->default_ruleset_id);
+        $statisticsRelationName = User::statisticsRelationName($ruleset);
+        if ($statisticsRelationName === null) {
+            throw new InvariantException(osu_trans('beatmaps.invalid_ruleset'));
+        }
+        $leaderboard = $team
+            ->members
+            ->loadMissing("user.{$statisticsRelationName}")
```

Comment:
also need to load/set the team and userGroups

Reply:
more like too much copypaste 👀